test/jdk/java/awt/EventDispatchThread/InterruptEDTTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2010, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Robot;
+import java.awt.Toolkit;
+import java.awt.event.InputEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+
+/*
+ * @test
+ * @key headful
+ * @bug 7184401
+ * @summary Verify that events are not lost when EDT is interrupted.
+ * @modules java.desktop/sun.awt
+ * @run main InterruptEDTTest
+ */
+
+public class InterruptEDTTest {
+
+    private static Frame frame = null;
+    private static volatile Thread edt = null;
+    private static volatile Robot robot = null;
+    private static volatile int xLocation;
+    private static volatile int yLocation;
+    private static volatile int width;
+    private static volatile int height;
+
+    public static void main(String args[]) throws Exception {
+        try {
+            robot = new Robot();
+            EventQueue.invokeAndWait(() -> {
+                edt = Thread.currentThread();
+                frame = new Frame("Frame");
+                frame.addMouseListener(new MouseAdapter() {
+                    public void mouseClicked(MouseEvent me) {
+                        System.out.println("Mouse clicked Event: " + me);
+                    }
+                });
+                frame.setBounds(350, 50, 400, 400);
+                frame.setLocationRelativeTo(null);
+                frame.setVisible(true);
+            });
+            ((sun.awt.SunToolkit) (Toolkit.getDefaultToolkit())).realSync();
```

Comment:
There was a comment in the original code before the realSync() call that was removed, which perhaps should have been left as is. 
        // cannot substitute with robot.waitForIdle() presumably because of flushPendingEvents() there
My understanding of above is that for simulating the test scenario - which is - Events (on line 75,76,77) are triggered and while these are potentially not yet complete(as they are wrapped in an invokeLater), the edt is interrupted. 

When run on jdk7u6  - the UI hangs and SOP of line 59 is not invoked. 
When run on jdk7u361 b01 the SOP on line 59 is printed and also the test passed.

waitForIdle here will trigger the flushPendingEvents internally before the edt interrupt call -and that would eliminate any possibility of simulating the above scenario.

Reply:
Then please double-check how robot.waitForIdle() and realSync() affect the test, since both call the flushPendingEvents(), why there is a difference?
 * robot.waitForIdle()->SunToolkit.flushPendingEvents();
 * realSync()->waitForIdle()->flushPendingEvents

Reply:
That's right. If the test hangs when `Robot.waitForIdle` is called, it looks like another problem in the JDK.

`Robot.waitForIdle` calls `SunToolkit.flushPendingEvents` followed by `SunToolkit.realSync`.

`SunToolkit.realSync`, in its turn, calls `SunToolKit.waitForIdle` in a loop which calls `SunToolkit.flushPendingEvents`.

It looks like a problem to me.